src/Illuminate/Foundation/Console/EnvironmentDiffCommand.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Illuminate\Foundation\Console;
+
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\File;
+use Symfony\Component\Console\Attribute\AsCommand;
+
+#[AsCommand(name: 'env:diff')]
+class EnvironmentDiffCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'env:diff
+                    {base? : The base environment file to compare against (default: .env.example)}
+                    {compare? : The environment file to compare (default: .env)}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Compare environment files and show differences';
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $baseFile = $this->argument('base') ?? '.env.example';
+        $compareFile = $this->argument('compare') ?? '.env';
+
+        $basePath = base_path($baseFile);
+        $comparePath = base_path($compareFile);
+
+        if (! File::exists($basePath)) {
+            $this->error("Base file '{$baseFile}' does not exist.");
+
+            return 1;
+        }
+
+        if (! File::exists($comparePath)) {
+            $this->error("Compare file '{$compareFile}' does not exist.");
+
+            return 1;
+        }
+
+        $baseEnv = $this->parseEnvFile($basePath);
+        $compareEnv = $this->parseEnvFile($comparePath);
+
+        $this->displayDifferences($baseEnv, $compareEnv, $baseFile, $compareFile);
+
+        return 0;
+    }
+
+    /**
+     * Parse an environment file and return an array of key-value pairs.
+     *
+     * @param  string  $filePath
+     * @return array
+     */
+    protected function parseEnvFile(string $filePath): array
```

Comment:
This parsing is incorrect. One should instead use the phpdotenv parser (or even a higher level component to do this).